st2stream/st2stream/controllers/v1/executions.py

```diff
@@ -0,0 +1,141 @@
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+
+import six
+
+from st2api.controllers.resource import ResourceController
+from st2common import log as logging
+from st2common.constants import action as action_constants
+from st2common.models.db.execution import ActionExecutionOutputDB
+from st2common.models.api.execution import ActionExecutionAPI
+from st2common.models.api.execution import ActionExecutionOutputAPI
+from st2common.persistence.execution import ActionExecution
+from st2common.persistence.execution import ActionExecutionOutput
+from st2common.router import Response
+from st2common.util.jsonify import json_encode
+from st2common.rbac.types import PermissionType
+from st2common.stream.listener import get_listener
+
+__all__ = [
+    'ActionExecutionOutputStreamController'
+]
+
+LOG = logging.getLogger(__name__)
+
+
+class ActionExecutionOutputStreamController(ResourceController):
+    model = ActionExecutionAPI
+    access = ActionExecution
+
+    supported_filters = {
+        'output_type': 'output_type'
+    }
+
+    CLOSE_STREAM_LIVEACTION_STATES = action_constants.LIVEACTION_COMPLETED_STATES + [
+        action_constants.LIVEACTION_STATUS_PAUSING,
+        action_constants.LIVEACTION_STATUS_RESUMING
+    ]
+
+    def get_one(self, id, output_type='all', requester_user=None):
+        # Special case for id == "last"
+        if id == 'last':
+            execution_db = ActionExecution.query().order_by('-id').limit(1).first()
+
+            if not execution_db:
+                raise ValueError('No executions found in the database')
+
+            id = str(execution_db.id)
+
+        execution_db = self._get_one_by_id(id=id, requester_user=requester_user,
+                                           permission_type=PermissionType.EXECUTION_VIEW)
+        execution_id = str(execution_db.id)
+
+        query_filters = {}
+        if output_type and output_type != 'all':
+            query_filters['output_type'] = output_type
+
+        def format_output_object(output_db_or_api):
+            if isinstance(output_db_or_api, ActionExecutionOutputDB):
+                data = ActionExecutionOutputAPI.from_model(output_db_or_api)
+            elif isinstance(output_db_or_api, ActionExecutionOutputAPI):
+                data = output_db_or_api
+            else:
+                raise ValueError('Unsupported format: %s' % (type(output_db_or_api)))
+
+            event = 'st2.execution.output__create'
```

Comment:
btw, can we rename it to something more appropriate? The verb `create` does not really represent what's happening here, IMHO.

Reply:
I used this name for consistency since it's the same name which is used for those events in the `/v1/stream` endpoint.